File: Module.Persist.TPM/Model/History/HistoricalPlanPostPromoEffect.cs
using System;
using Core.History;
using Module.Persist.TPM.Model.TPM;

namespace Module.Persist.TPM.Model.History
{
    [AssociatedWith(typeof(PlanPostPromoEffect))]
    public class HistoricalPlanPostPromoEffect: BaseHistoricalEntity<Guid>
    {
        public int? ClientTreeId { get; set; }
        public Guid? BrandTechId { get; set; }
        public string Size { get; set; }
        public double PlanPostPromoEffectW1 { get; set; }
        public double PlanPostPromoEffectW2 { get; set; }
        public string DiscountRangeName { get; set; }
        public string DurationRangeName { get; set; }

        private string brandTechName;
        public string BrandTechName
        {
            get
            {
                return string.IsNullOrEmpty(BrandTechBrandsegTechsub)
                    ? brandTechName
                    : BrandTechBrandsegTechsub;
            }
            set
            {
                brandTechName = value;
            }
        }
        public string BrandTechBrandsegTechsub { get; set; }
        public string ClientTreeFullPathName { get; set; }
        public int? ClientTreeObjectId { get; set; }
    }
}

File: Module.Frontend.TPM/Client/model/tpm/planpostpromoeffect/HistoricalPlanPostPromoEffect.js
Ext.define('App.model.tpm.planpostpromoeffect.HistoricalPlanPostPromoEffect', {
    extend: 'Ext.data.Model',
    idProperty: '_Id',
    breezeEntityType: 'PlanPostPromoEffect',
    fields: [
        { name: '_Id', type: 'string', hidden: true },
        { name: '_ObjectId', hidden: true },
        { name: '_User', type: 'string', isDefault: true },
        { name: '_Role', type: 'string', isDefault: true },
        { name: '_EditDate', type: 'date', isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: '_Operation', type: 'string', isDefault: true },
        { name: 'BrandTechId', hidden: true, isDefault: true },
        { name: 'BrandTechName', type: 'string', useNull: true, hidden: false, isDefault: true },
        { name: 'ClientTreeId', hidden: true, isDefault: true },
        { name: 'ClientTreeFullPathName', type: 'string', hidden: false, isDefault: true },
        { name: 'ClientTreeObjectId', type: 'int', hidden: false, isDefault: true },
        { name: 'Size', type: 'string', hidden: false, isDefault: true },
        { name: 'PlanPostPromoEffectW1', type: 'float', hidden: false, isDefault: true, },
        { name: 'PlanPostPromoEffectW2', type: 'float', hidden: false, isDefault: true, },
        { name: 'DiscountRangeId', hidden: true, useNull: true, isDefault: true },
        { name: 'DiscountRangeName', type: 'string', useNull: true, hidden: false, isDefault: true },
        { name: 'DurationRangeId', hidden: true, useNull: true, isDefault: true },
        { name: 'DurationRangeName', type: 'string', useNull: true, hidden: false, isDefault: true }
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'HistoricalPlanPostPromoEffects',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        },
        extraParams: {
            Id: null
        }
    }
});
